tests/unit/serve/executors/test_executor.py

```diff
@@ -435,3 +444,62 @@ def test_to_docker_compose_yaml(tmpdir, exec_type):
             assert services['gateway']['ports'][0] == '2020:2020'
             gateway_args = services['gateway']['command']
             assert gateway_args[gateway_args.index('--port') + 1] == '2020'
+
+
+def _create_test_data_message(counter=0):
+    return list(request_generator('/', DocumentArray([Document(text=str(counter))])))[0]
+
+
+@pytest.mark.asyncio
+async def test_blocking_sync_exec():
+    SLEEP_TIME = 0.01
+    REQUEST_COUNT = 100
+
+    class BlockingExecutor(Executor):
+        @requests
+        def foo(self, docs: DocumentArray, **kwargs):
+            time.sleep(SLEEP_TIME)
+            for doc in docs:
+                doc.text = 'BlockingExecutor'
+            return docs
+
+    args = set_pod_parser().parse_args(['--uses', 'BlockingExecutor'])
+
+    cancel_event = multiprocessing.Event()
+
+    def start_runtime(args, cancel_event):
+        with WorkerRuntime(args, cancel_event) as runtime:
+            runtime.run_forever()
+
+    runtime_thread = Process(
+        target=start_runtime,
+        args=(args, cancel_event),
+        daemon=True,
+    )
+    runtime_thread.start()
+
+    assert AsyncNewLoopRuntime.wait_for_ready_or_shutdown(
+        timeout=5.0,
+        ctrl_address=f'{args.host}:{args.port}',
+        ready_or_shutdown_event=Event(),
+    )
+
+    send_tasks = []
+    start_time = time.time()
+    for i in range(REQUEST_COUNT):
+        send_tasks.append(
+            asyncio.create_task(
+                GrpcConnectionPool.send_request_async(
+                    _create_test_data_message(),
+                    target=f'{args.host}:{args.port}',
+                    timeout=3.0,
+                )
+            )
+        )
+
+    await asyncio.gather(*send_tasks)
```

Comment:
should we also `assert` that the requests finish successfully? Check the responses returned in every request?